lib/aio/job.py

```diff
@@ -0,0 +1,190 @@
+# Copyright (C) 2024 Red Hat, Inc.
```

Comment:
What's the rationale of splitting this out? This is core functionality of `job-runner`, and not being imported anywhere else. There also isn't a unit test for it, nor does it seem realistic to have one.

Reply:
I want to unit test this stuff and I also want to call it from a future amqp-fed mode (which will likely live in a separate file).